core/src/main/java/com/mongodb/stitch/core/internal/net/StitchRequestClient.java

```diff
@@ -27,11 +38,39 @@ private static Response inspectResponse(final Response response) {
   }
 
   public Response doRequest(final StitchRequest stitchReq) {
+    ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
I'm concerned about this line in particular. Is there going to be significant overhead in creating a thread that will only live for a single network request? Would it be better if this request executor thread was a member of the request client so it's shared among requests? Or even perhaps to have a thread pool?

Reply:
I think having this `ExecutorService` be an instance or static member of `StitchRequestClient` would be most similar to what I have in the Swift SDK (as the PR currently stands) but I'm open to suggestions.

Reply:
Many threads are being spun up for various tasks during the lifecycle of any app. However, I'm not sure this is how we should be doing it, but not because of overhead. 

Either we just spin up a single thread and don't bother with executors, or spin up a thread pool.